raiden/network/proxies/user_deposit.py

```diff
@@ -41,3 +165,99 @@ def effective_balance(self, address: Address, block_identifier: BlockSpecificati
             raise RuntimeError(f"Call to 'effectiveBalance' returned nothing")
 
         return balance
+
+    def _deposit_preconditions(
+            self,
+            total_deposit: TokenAmount,
+            beneficiary: Address,
+            token: Token,
+            block_identifier: BlockSpecification,
+    ) -> Tuple[TokenAmount, Dict]:
+        if not isinstance(total_deposit, int):
+            raise ValueError('total_deposit needs to be an integral number.')
+
+        previous_total_deposit = self.get_total_deposit(
+            address=beneficiary,
+            block_identifier=block_identifier,
+        )
+        amount_to_deposit = total_deposit - previous_total_deposit
+
+        log_details = {
+            'user_deposit_address': pex(self.address),
+            'node': pex(self.node_address),
+            'beneficiary': pex(beneficiary),
+            'new_total_deposit': total_deposit,
+            'previous_total_deposit': previous_total_deposit,
+        }
+
+        if total_deposit < previous_total_deposit:
+            msg = (
+                f'Current total deposit ({previous_total_deposit}) is already larger '
+                f'than the requested total deposit amount ({total_deposit})'
+            )
+            log.info('deposit failed', reason=msg, **log_details)
+            raise DepositMismatch(msg)
+
+        if amount_to_deposit <= 0:
+            msg = (
+                f'new_total_deposit - previous_total_deposit must be greater than 0. '
+                f'new_total_deposit={total_deposit} '
+                f'previous_total_deposit={previous_total_deposit}'
+            )
+            log.info('deposit failed', reason=msg, **log_details)
+            raise DepositMismatch(msg)
+
+        current_balance = token.balance_of(self.node_address)
```

Comment:
you should get the balance at the consistent blockchain view of the provided block_identifier.